Controller/FormInteractionController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Controller;
+
+use \OxidEsales\Eshop\Application\Controller\FrontendController;
+use \OxidEsales\Eshop\Core\Registry;
+
+use \Wirecard\Oxid\Model\FormInteractionResponseFields;
+use \Wirecard\PaymentSdk\Entity\FormFieldMap;
+
+/**
+ * Class Form_Interaction_Controller
+ */
+class FormInteractionController extends FrontendController
+{
+    /**
+     * @inheritdoc
+     */
+    protected $_sThisTemplate = 'form_interaction.tpl';
+
+    /**
+     * @var FormInteractionResponseFields
+     */
+    private $_oResponse = null;
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+
+        $this->_oResponse = Registry::getSession()->getVariable('wdFormInteractionResponse');
+        if (empty($this->_oResponse)) {
+            //redirect home
+            Registry::getUtils()->redirect(Registry::getConfig()->getShopUrl() . 'index.php');
```

Comment:
There is getShopHomeUrl() method in config. Could be used here?

Reply:
nice! thanks for the hint!